<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ExpressionCreatorDialog</class>
 <widget class="QDialog" name="ExpressionCreatorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>542</width>
    <height>734</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Expression Creator</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="labelVariables">
         <property name="text">
          <string>Variables:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="listViewVariableBox">
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="labelOperators">
         <property name="text">
          <string>Operators:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="listViewOperatorBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="sortingEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelConditionalOperators">
         <property name="text">
          <string>Conditional Operators</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="listViewConditionalBox">
         <property name="sortingEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="labelSelectedVariableDescription">
     <property name="text">
      <string>Variable Description...</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="layoutExpression">
     <item>
      <widget class="QLabel" name="labelExpression">
       <property name="text">
        <string>Expression:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEditExpression">
       <property name="text">
        <string/>
       </property>
       <property name="placeholderText">
        <string>volume * 4 / 10</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="labelErrorExpression">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Error&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="layoutCheckboxes">
     <item>
      <widget class="QCheckBox" name="checkBoxAddToHeaders">
       <property name="text">
        <string>Add to Headers</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBoxSaveVariable">
       <property name="text">
        <string>Save as Custom Variable</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="layoutHeader">
     <item>
      <widget class="QLabel" name="labelHeaderName">
       <property name="text">
        <string>Header Name:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEditHeaderName">
       <property name="text">
        <string/>
       </property>
       <property name="placeholderText">
        <string>Custom Volume</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelErrorHeader">
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Error&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="layoutConditional">
     <item>
      <widget class="QCheckBox" name="checkBoxAddConditional">
       <property name="text">
        <string>Add Conditional</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelConditional">
       <property name="text">
        <string>Conditional:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEditConditional">
       <property name="text">
        <string/>
       </property>
       <property name="placeholderText">
        <string>if volume &gt; 100 then Color.Red</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelErrorConditional">
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Error&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="layoutCustomVariables">
     <item>
      <layout class="QVBoxLayout" name="layoutNameSubCustomVariables">
       <item>
        <widget class="QLabel" name="labelCustomVariable">
         <property name="text">
          <string>Custom Variable Name</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEditCustomVariableName">
         <property name="placeholderText">
          <string>newVariable</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="labelErrorVariableName">
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Error&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="layoutDescriptionSubCustomVariables">
       <item>
        <widget class="QLabel" name="labelDescription">
         <property name="text">
          <string>Description:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEditVariableDescription">
         <property name="placeholderText">
          <string>volume multiplied by 4 and divided by 10</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ExpressionCreatorDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>631</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ExpressionCreatorDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>637</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
